### PR TITLE
[FIX] Import Images: case sensitivity and remove unused code

### PR DESCRIPTION
**Pull request by jerneju**
case sensitivity and remove unused code

### CODE REVIEW

File: orangecontrib/imageanalytics/widgets/owimageimport.py [hunk 1]
```diff
@@ -695,20 +695,15 @@ def __init__(self, startdir, formats=DefaultFormats, report_progress=None):
 
     def run(self):
         imgmeta = []
-        filescanner = scan(self.startdir)
         patterns = ["*.{}".format(fmt.lower()) for fmt in self.formats]
-
-        def fnmatch_any(fname, patterns):
-            return any(fnmatch.fnmatch(fname.lower(), pattern)
-                       for pattern in patterns)
+        filescanner = scan(self.startdir, include_patterns=patterns)
```
Comment (ales-erjavec) [inline orangecontrib/imageanalytics/widgets/owimageimport.py]: the `fnmatch_any` forced case insensitive matching (in combination with `fmt.lower()` few lines up). This is not retained here.

Comment (jerneju) [inline orangecontrib/imageanalytics/widgets/owimageimport.py]: Ok, thanks, fixed.

Comment (jerneju) [inline orangecontrib/imageanalytics/widgets/owimageimport.py]: Can we merge this?

Comment (ales-erjavec) [reply orangecontrib/imageanalytics/widgets/owimageimport.py]: Now `scan` no longer does what it claims.

```sh
touch foo.BAR
```
```python
next(iter(scan("./", include_patterns=["*.BAR"])))
```

Can you add a `case_insensitive` parameter to scan
```
case_insensitive : bool
   Ignore case in both filename(s) and pattern(s).

**Comment by ales-erjavec** [comment]
Looking at the code again, I see there is a very good reason for the current implementation. The loop in `ImageScan.run` allows cancellation at all `os.walk` paths encountered, regardless if they match the pattern. In the opposite case (as proposed here) there is a risk the user runs the scan in a directory containing lots (say millions) of non matching paths and the execution flow would not return from the `scan` until all are traversed.

**Pull request merged**